### PR TITLE
Add a Flag for Weeks

### PR DESCRIPTION
**Pull request by athul**
Due to Feedback, I guess the week in the Top should be made optional since it could increase the clutter in the readme. This patch is a fix for that. i have to test it a bit and shouldn't be merged till then

### CODE REVIEW

**Comment by athul** [comment]
@gautamkrishnar and @joe733 you reviews and feedbacks please

File: README.md [hunk 1]
```diff
@@ -95,3 +95,18 @@ jobs:
...
+
+1. If you want to add the week in the Header of your stats, you can add `HEAD_FLAG: true` in your workflow file like this
+
+```yml
+- uses: athul/waka-readme@master
+        with:
+          WAKATIME_API_KEY: ${{ secrets.WAKATIME_API_KEY }}
+          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+          USERNAME: <username>
+          HEAD_FLAG: true
```
Comment (gautamkrishnar) [inline README.md:109]: Can we rename the env to something descriptive like `SHOW_TITLE`

File: action.yml [hunk 2]
```diff
@@ -15,7 +15,12 @@ inputs:
   USERNAME:
     description: 'Your GitHub username'
     default: ${{ github.repository_owner }}
+    required: true
 
+  WEEK_FLAG:
```
Comment (gautamkrishnar) [inline action.yml:20]: Why this is `WEEK_FLAG` here and `HEAD_FLAG` in the code?

File: action.yml [hunk 3]
```diff
@@ -15,7 +15,12 @@ inputs:
   USERNAME:
     description: 'Your GitHub username'
     default: ${{ github.repository_owner }}
+    required: true
```
Comment (gautamkrishnar) [inline action.yml:18]: Do we need this change, making username required?

File: README.md [hunk 4]
```diff
@@ -95,3 +95,18 @@ jobs:
...
+```
+
+`HEAD_FLAG` is defaulted to `false` and can be overriden by `true`
```
Comment (gautamkrishnar) [inline README.md:112]:
```
`SHOW_TITLE` flag can be set to true if you want to display the week number and days in the readme
```

File: action.yml (hunk 2 above)
Comment (athul) [reply action.yml:20]: Lol, I was in a rush 😂

Comment (gautamkrishnar) [reply action.yml:20]: 😄  hehhe chill bro

File: main.py [hunk 5]
```diff
@@ -46,7 +47,10 @@ def get_stats():
         op = f"{l['name']}{' '*(12-ln)}{l['text']}{' '*(20-ln_text)}{make_graph(l['percent'])}   {l['percent']}%"
         data_list.append(op)
     data = ' \n'.join(data_list)
-    return '```text\n'+this_week()+'\n\n'+data+'\n```'
+    if head_flag is False:
+        return '```text\n'+this_week()+'\n\n'+data+'\n```'
+    else:
+        return return '```text\n\n'+data+'\n```'
```
Comment (athul) [inline main.py:53]: Lol, how dumb am I 🤪

Comment (gautamkrishnar) [reply main.py:53]: Oh! recursion 😮

File: README.md [hunk 6]
```diff
@@ -106,7 +106,17 @@ jobs:
           WAKATIME_API_KEY: ${{ secrets.WAKATIME_API_KEY }}
           GH_TOKEN: ${{ secrets.GH_TOKEN }}
           USERNAME: <username>
-          HEAD_FLAG: true
+          SHOW_TITLE: true
```
Comment (joe733) [inline README.md:109]: Will it be good if:
```yml
SHOW_TITLE: true
```
was set as `false` in the template above Like this?

```yml
name: Waka Readme
on:
  schedule:
    # Runs at 12am UTC
    - cron: '0 0 * * *'
jobs:
  update-readme:
    name: Update this repo's README
    runs-on: ubuntu-latest
    steps:
      - uses: athul/waka-readme@master
        with:
          WAKATIME_API_KEY: ${{ secrets.WAKATIME_API_KEY }}
          SHOW_TITLE: false # default | optional - refer the Extras section
```

So they just have to toggle `true` / `false`

File: main.py [hunk 7]
```diff
@@ -17,12 +17,16 @@
...
-    week_end = datetime.datetime.today() - datetime.timedelta(days=1)
-    week_start = week_end - datetime.timedelta(days=7)
-    return f"Week: {week_start.strftime('%d %B, %Y')} - {week_end.strftime('%d %B, %Y')}"
+    '''Returns current week span'''
+    week_number = datetime.date.today().isocalendar()[1]
+    month = datetime.date.today().strftime('%B')
+    week_start = datetime.datetime.today().day - datetime.datetime.today().weekday()
+    week_end = week_start + 5
+    print("Week's header Created")
+    return f"Week #{week_number} : {month} {week_start} - {week_end}"
```
Comment (joe733) [inline main.py:29]: No, wait this will resurface the #7 ... right?

File: main.py [hunk 8]
```diff
@@ -41,11 +45,16 @@ def get_stats():
     for l in lang_data[:5]:
         ln = len(l['name'])
         ln_text = len(l['text'])
-        fmt_percent = format(l['percent'], '0.2f').zfill(5) # to provide a neat finish.
-        data_list.append(
-            f"{l['name']}{' '*(12-ln)}{l['text']}{' '*(20-ln_text)}{make_graph(l['percent'])}   {fmt_percent} %")
+        op = f"{l['name']}{' '*(12-ln)}{l['text']}{' '*(20-ln_text)}{make_graph(l['percent'])}   {l['percent']}%"
```
Comment (joe733) [inline main.py:48]: Opps! I think you removed the pretty formatting. This is from the old file...

Comment (athul) [reply main.py:48]: Oh Damn🤦‍♂️

File: README.md (hunk 6 above)
Comment (gautamkrishnar) [reply README.md:109]: Since its optional flasg its okay to omit it, but this example is a little more explanatory.

**Comment by athul** [comment]
I'll just send a force pushed update

**Comment by athul** [comment]
@gautamkrishnar and @joe733 Now

File: main.py [hunk 9]
```diff
@@ -44,17 +46,23 @@ def get_stats():
...
 
-def decode_readme(data: str):
+def decode_readme(data: str) -> str:
     '''Decode the contets of old readme'''
     decoded_bytes = base64.b64decode(data)
     return str(decoded_bytes, 'utf-8')
 
 
-def generate_new_readme(stats: str, readme: str):
+def generate_new_readme(stats: str, readme: str) ->str:
```
Comment (joe733) [inline main.py:65]: What is this `-> str`? return type?

Comment (joe733) [reply main.py:65]: Cool! I didn't know [this](https://stackoverflow.com/questions/14379753) feature existed!

Comment (gautamkrishnar) [reply main.py:65]: Yea me too. Good to know.

Comment (athul) [reply main.py:65]: They're function Annotations. Not exactly return type but serves the purpose

**Pull request merged**